### PR TITLE
Beam Devastator (RPG Replacement) and Beam Cannon (research)

### PR DESCRIPTION
**Pull request by Stop-Signs**
## About the PR
Adds the beam devastator, a much more powerful version of the beam cannon in #2221. Since the RPG is on the chopping block this could serve as a worthy replacement. It shoots several small damage beams a second and can be sprayed effectively. mag size of 500. takes incredibly long to recharge (about 10 minutes with a standard recharger), however it can be slotted into a cage recharger which has also been buffed, at this rate it takes ~3 minutes to fully charge.

#2221 was closed as this PR adds it anyway, adds the beam cannon as well to the sec research tree

IMPORTANT NOTE!!!!!!:  This item is UNOBTAINABLE and in order to replace the RPG will need to be MAPPED/MIGRATED to do so. I am willing to map it myself but I don't exactly know how to do so.

## Why / Balance
Curators were looking for an RPG replacement, this gun has a fairly low TTK and instead relies on sustained fire to be as oppressively as it is. A worthy successor to the RPG while not being just a no-skill one-tapinator

## Media
Github says the videos are too big. heres the sprite instead
base

## Requirements
- [x] I have tested all added content and changes.
- [x] I have added media to this PR or it does not require an ingame showcase.>

:cl:
- add: Added the new Beam Cannon for Epi to research.
- tweak: The Cage Recharger can now charge lasers.

### CODE REVIEW

**Comment by Lyndomen** [comment]
https://github.com/user-attachments/assets/99be5a2c-51a6-4419-9512-a52eea683a23

Both of these weapons are balanced by their frankly **bad** time to kill, being about twice that of a Universal. To compensate, they have massive ammo capacities and are intended to hold down an area with suppressing fire. 

The projectile sprite itself is sped up on its frame time to feel more snappy, and feel more like a "beam". Similarly, the sound is a bass-y hum rather than a sharp pop- to accommodate for the hearing of everyone else.

**Comment by ThataKat** [comment]
Makes sense for security to have a "heavy" weapon that's more defensive (doesn't literally punch a hole in their own vessel).

**Comment by Lyndomen** [comment]
> Makes sense for security to have a "heavy" weapon that's more defensive (doesn't literally punch a hole in their own vessel).

For direction's discussion, the Devestator **does do structural damage.** 
image

However, its still a laser and infinity more controllable. It also takes a GOOD bit of time and ammo, but is genuinely feasible. This gives security a reliable "breaching" weapon as long as what they are breaching into isn't a window :trollface:  

I intend to PR a researchable "mining/deconstruction" version of this gun eventually.

**Comment by Lyndomen** [comment]
Also quick note that walls & structures reduce Structural damage by 10, and then another percentage, so "20 structural" is a lot less than 30. It takes time to breach, I recommend playing with the gun in a sandbox.

**Comment by Radezolid** [comment]
I think it isn't a good replacement for the RPG, for example a nuclear operative can just buy a E-Shield to counter the devastator and forget about it.
Besides that, i love the guns.

**Comment by Lyndomen** [comment]
> I think it isn't a good replacement for the RPG, for example a nuclear operative can just buy a E-Shield to counter the devastator and forget about it. Besides that, i love the guns.

Real, I was thinking about reducing shield reflect chance and buffing it's health value anyway

**Comment by Stop-Signs** [comment]
it takes ~100 shots to bust a reinforced wall

**Comment by Radezolid** [comment]
Also, it won't really help as a breaching tool unless you're breaching an airlock since the lasers go thru the girders + the slowdown, you can just use a breaching charge instead

**Comment by Lyndomen** [comment]
Requesting review, should be good to go honestly as it's balance approved & the RPG replacement happens per map.

**Comment by CybersunOperative** [comment]
RPG REPLACEMENT REAL RAAAAGH.
I like it. It feels more suppressive instead of "Fire and Forget" antag obliterator. Zombie Mains weeping. 

This sentiment is shared by direction. (Who also hate the RPG) :blunt:
I would ask mappers after this PR is merged about this change.

**Comment by LadyDanger2743** [comment]
Really the only thing that I'm curious about is how effective it is at clearing standard and reinforced walls. Other than that, it has the Direction Seal Of Approval.

**Comment by Stop-Signs** [comment]
> Really the only thing that I'm curious about is how effective it is at clearing standard and reinforced walls. Other than that, it has the Direction Seal Of Approval.

I did test this, it takes over 100 shots to do so!

**Comment by LadyDanger2743** [comment]
> > Really the only thing that I'm curious about is how effective it is at clearing standard and reinforced walls. Other than that, it has the Direction Seal Of Approval.
> 
> I did test this, it takes over 100 shots to do so!

Alright, that's all I really need. This has Direction's approval to merge.

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml]: prototype order

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml]: above

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml]: spaces in inline lists

File: Resources/Prototypes/Entities/Structures/Power/chargers.yml
Comment (MilonPL) [inline Resources/Prototypes/Entities/Structures/Power/chargers.yml]: comments

File: Resources/Prototypes/Entities/Structures/Power/chargers.yml
Comment (MilonPL) [inline Resources/Prototypes/Entities/Structures/Power/chargers.yml]: above

File: Resources/Prototypes/Research/arsenal.yml
Comment (MilonPL) [inline Resources/Prototypes/Research/arsenal.yml]: why, just do a single comment

**Review by MilonPL — CHANGES_REQUESTED** [review]

**Comment by therealDLondon** [comment]
This isnt code approval, but its approved as far as im concerned.
Direction approved, millions must merge

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml]: not a proper noun, beam cannon

File: Resources/Prototypes/Entities/Structures/Power/chargers.yml
Comment (deltanedas) [inline Resources/Prototypes/Entities/Structures/Power/chargers.yml:142]: blacklist power cells too or this is just gamer turbo charger

File: Resources/Textures/DeltaV/Objects/Weapons/Guns/Battery/beam_cannon.rsi/meta.json
Comment (deltanedas) [inline Resources/Textures/DeltaV/Objects/Weapons/Guns/Battery/beam_cannon.rsi/meta.json]: fix indent

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml]: unindent this by 2 spaces :trollface:

File: Resources/Textures/DeltaV/Objects/Weapons/Guns/Battery/beam_cannon.rsi/meta.json
Comment (deltanedas) [inline Resources/Textures/DeltaV/Objects/Weapons/Guns/Battery/beam_cannon.rsi/meta.json]: this is still 2 spaced

**Review by deltanedas — APPROVED** [review]

**Comment by deltanedas** [comment]
removed the beam devastator line from cl as this pr doesnt replace anything and it isnt likely to get migrated for every map

**Pull request merged**
